### PR TITLE
feat(agents): add ant agent

### PR DESCRIPTION
**Pull request by muchvo**
## Description

1. Add ant agent.

## Motivation and Context

Why is this change required? What problem does it solve?
If it fixes an open issue, please link to the issue here.
You can use the syntax `close #15213` if this solves the issue #15213

- [ ] I have raised an issue to propose this change ([required](https://github.com/PKU-MARL/omnisafe/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Implemented Tasks

- [ ] Subtask 1
- [ ] Subtask 2
- [ ] Subtask 3

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://omnisafe.readthedocs.io/en/latest/developer/contributing.html) guide. (**required**)
- [x] My change requires a change to the documentation.
- [x] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.
- [ ] I have reformatted the code using `make format`. (**required**)
- [ ] I have checked the code using `make lint`. (**required**)
- [ ] I have ensured `make test` pass. (**required**)

### CODE REVIEW

**Review by zmsn-2077 — APPROVED** [review]
After careful review, I agree with these changes.

File: envs/safety-gymnasium/safety_gymnasium/__init__.py [hunk 1]
```diff
@@ -16,67 +16,33 @@
 
 from copy import deepcopy
 
-from safety_gymnasium import vector
 from safety_gymnasium.utils.registration import make, register
-from safety_gymnasium.version import __version__
 
 
-__all__ = ['register', 'make', 'vector']
```
Comment (zmsn-2077) [inline envs/safety-gymnasium/safety_gymnasium/__init__.py:24]: Will this change affect the use of `safety_gymnasium.vector.make`?

**Review by zmsn-2077 — APPROVED** [review]
The PR is very clean, and I agree with these changes.

File: envs/safety-gymnasium/safety_gymnasium/agents/ant.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    """
+
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        random_generator: RandomGenerator,
+        placements: list = None,
+        locations: list = None,
+        keepout: float = 0.4,
+        rot: float = None,
+    ):
```
Comment (Gaiejj) [inline envs/safety-gymnasium/safety_gymnasium/agents/ant.py:36]: missing docstring

**Review by Gaiejj — DISMISSED** [review]

**Review by zmsn-2077 — APPROVED** [review]

**Review by Gaiejj — APPROVED** [review]

**Pull request merged**